### PR TITLE
Updated Algorithm for Assigning Houses

### PR DESCRIPTION
**Pull request by RiceWater**
- Updated the random house assignment generator to ensure that the population among different houses will be equal

### CODE REVIEW

File: parse-server-ascension/cloud/student.js [hunk 1]
```diff
@@ -187,20 +187,41 @@ Parse.Cloud.define("AssignHouse", async(request) => {
...
         "StudentID" : argument.StudentID,
-        "HouseID" : pickedHouse.id,
+        "HouseID" : res2[idx].objectId,
     }
 
     await Parse.Cloud.run("AddHouseMember", params);
     console.log("Successfully Assigned House");
 });
 
+function getRndInteger(min, max) {  //min = inclusive; max = exclusive
```
Comment (aid8) [inline parse-server-ascension/cloud/student.js:221]: This should be in global.js

Comment (RiceWater) [reply parse-server-ascension/cloud/student.js:221]: All right

File: parse-server-ascension/cloud/student.js [hunk 2]
```diff
@@ -187,20 +187,41 @@ Parse.Cloud.define("AssignHouse", async(request) => {
...
-    
-    //Get lowest!
-    var randNum = Math.floor(Math.random() * res.length);
-    var pickedHouse = res[randNum];
 
+    //Note: All houses should begin with the same population count; base case (e.g [0,0,0,0])
+    const res2 = JSON.parse(JSON.stringify(res))
+    let max = 0
+    let counter = 0
+    for(let i = 0; i < res2.length; i++){
```
Comment (aid8) [inline parse-server-ascension/cloud/student.js:195]: You can use prebuilt functions in js

**Example**
var nums = [1, 2, 3]
Math.min.apply(Math, nums); //returns 1
Math.max.apply(Math, max); //returns 3

File: parse-server-ascension/cloud/student.js [hunk 3]
```diff
@@ -187,20 +187,41 @@ Parse.Cloud.define("AssignHouse", async(request) => {
...
+            counter++
+        }
+    }
+    //counter == 0 => houses have same population [1,1,1,1]
+    //counter == 1 => houses have 2 different populations [1,0,0,1]
+    //cuonter > 1 => houses have 3 or more different populations [1,3,0,1]
+    if(counter == 1){
+        for(let i = 0; i < res2.length; i++){
+            if(res2[i].HousePopulation == max){
+                res2.splice(i, 1)
```
Comment (aid8) [inline parse-server-ascension/cloud/student.js:207]: Not sure pag splice digdi, baka may ma miss out na index, safe na mag decrement ning i paka splice, syempre check mo rin baka mag negative si i.
Ignore nalang kung working naman.

**Safer/Efficient solution?** 
Instead na max, min nalang hanapon then insert the similar values in a new array, O(n) kasi si splice, and O(1) lang mag push

Comment (aid8) [reply parse-server-ascension/cloud/student.js:207]: Suggestion ko lang yan last comment haha

File: parse-server-ascension/cloud/student.js [hunk 4]
```diff
@@ -187,20 +187,41 @@ Parse.Cloud.define("AssignHouse", async(request) => {
...
+    for(let i = 0; i < res2.length; i++){
+        if(max < res2[i].HousePopulation){
+            max = res2[i].HousePopulation
+            counter++
+        }
+    }
+    //counter == 0 => houses have same population [1,1,1,1]
+    //counter == 1 => houses have 2 different populations [1,0,0,1]
+    //cuonter > 1 => houses have 3 or more different populations [1,3,0,1]
+    if(counter == 1){
```
Comment (aid8) [inline parse-server-ascension/cloud/student.js:204]: You forgot to change it to > 1, this will only check houses with 2 different populations right?

Comment (RiceWater) [reply parse-server-ascension/cloud/student.js:204]: Haha yeah, cause it'll filter out before there'll be 3 different populations, considering that we'll only have 4 houses

**Pull request merged**